cfy_cluster_setup/main.py

```diff
@@ -291,13 +298,57 @@ def generate_config(output_path):
                 output_path)
 
 
+def _check_path(config, key):
+    path = config.get(key)
+    if (not path) or (not exists(expanduser(path))):
+        raise ValidationError('Path {0} does not exist'.format(path))
+
+    config[key] = expanduser(path)
+
+
+def _check_value_supplied(config, key):
+    if not config.get(key):
+        raise ValidationError('{0} is not specified'.format(key))
+
+
+def _check_existing_vms(config):
+    ca_cert_path = config.get('ca_cert_path')
+    if ca_cert_path:
+        _check_path(config, 'ca_cert_path')
+
+    existing_vms_list = config.get('existing_vms')
+    for vm_name, vm_dict in existing_vms_list.items():
+        logger.info('Validating %s', vm_name)
+        if (not vm_dict['private_ip']) or (not vm_dict['public_ip']):
```

Comment:
I think we should maybe default public_ip to private_ip if it's not supplied, for environments where there's only one IP for cluster members?

Reply:
Yep, done for the first comment.